### ISSUE TITLE
geometry.transform: warp_perspective/warp_perspective3d warp float64 inputs through a float32 grid

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4776]
`warp_perspective` and `warp_perspective3d` build their sampling grid in float32 (`create_meshgrid(...)` / `create_meshgrid3d(...)` without `dtype`, cast afterwards), so a float64 input is warped through a float32 grid. An identity homography does not reproduce a float64 image to float64 roundoff; the error is about 1e-7 for an 8-pixel side and grows with the size. `warp_affine` / `warp_affine3d` are exact to float64 roundoff on the same input. `RandomPerspective3D` / `RandomPerspective` inherit the loss (with `distortion_scale=0`).

## Repro

```python
import torch, kornia
from kornia.geometry import warp_perspective, warp_perspective3d, warp_affine
print(kornia.__version__, torch.__version__)
torch.manual_seed(0)
for n in (8, 64):
    x = torch.rand(1, 1, n, n, dtype=torch.float64)
    I = torch.eye(3, dtype=torch.float64)[None]
    print(n, "warp_perspective", (warp_perspective(x, I, (n, n), align_corners=True) - x).abs().max().item(),
          "warp_affine", (warp_affine(x, I[:, :2], (n, n), align_corners=True) - x).abs().max().item())
    v = torch.rand(1, 1, n // 2, n, n, dtype=torch.float64)
    I4 = torch.eye(4, dtype=torch.float64)[None]
    print(n, "warp_perspective3d", (warp_perspective3d(v, I4, (n // 2, n, n), align_corners=True) - v).abs().max().item())
```

## Observed (kornia 0.9.0rc1 @ 57cd68e6f, torch 2.14.0, CPU)

```
8 warp_perspective 1.65e-07 warp_affine 0.0
8 warp_perspective3d 2.70e-07
64 warp_perspective 2.84e-06 warp_affine 9.88e-15
64 warp_perspective3d 3.57e-06
```

## Expected

A float64 input is warped through a float64 grid, so the identity warp reproduces it to float64 roundoff, as `warp_affine` does. `create_meshgrid` and `create_meshgrid3d` already take `dtype=`; the grid is built in float32 at `kornia/geometry/transform/imgwarp.py` (`warp_perspective`: `create_meshgrid(...).to(src.dtype)`; `homography_warp3d`: `create_meshgrid3d(...)` with no `dtype`). Half precision may keep building the normalized grid in float32 (#4231); the defect is the float64 case.

## Why this is a bug

kornia preserves the input dtype, and a float64 caller expects float64 precision; the affine siblings deliver it and the perspective ones silently do not. The `RandomPerspective3D` docstring currently documents the loss as a convention.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by ducha-aiki** [comment]
Note for the fix PR: `docs/source/get-started/conventions.rst` describes this defect without an issue anchor, so `grep -rnE "#4776|issues/4776"` misses it. The sentence is in the 3-D warning of the `align_corners` section: "The grid is built in ``float32``, so a ``float64`` volume is reproduced only to about ``1e-6`` even there". It has been on `main` since #4634. #4556 rewrites that warning and keeps a version of the sentence, so whichever of #4556 and the #4776 fix lands second has to remove or update it. The anchored carriers are the `warp_perspective`/`warp_perspective3d` Convention blocks (#4788) and the augmentation blocks (#4791).

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
